pkg/bmc/access.go

```diff
@@ -0,0 +1,88 @@
+package bmc
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
+// NOTE(dhellmann): This structure is very likely to change as we
+// adapt it to additional types.
+type AccessDetails struct {
+	bmcType         string
+	portNum         string
+	hostname        string
+	macAddress      string
+	physicalNetwork string
+}
+
+// NewAccessDetails creates an AccessDetails structure from the URL
+// for a BMC.
+func NewAccessDetails(address string) (*AccessDetails, error) {
+	parsedURL, err := url.Parse(address)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse BMC address information")
+	}
+
+	addr := &AccessDetails{
+		portNum:  parsedURL.Port(),
+		hostname: parsedURL.Hostname(),
+	}
+
+	if parsedURL.Scheme == "libvirt" {
+		addr.bmcType = "libvirt"
+	} else {
+		addr.bmcType = "ipmi"
+	}
+
+	// NOTE(dhellmann): libvirt-based hosts used for dev and testing
+	// require a MAC address and network name, specified as query
+	// parameters in the URL.
+	params := parsedURL.Query()
+	if len(params["mac"]) >= 1 {
+		// FIXME(dhellmann): There is probably a safer way to
```

Comment:
Here it is: https://golang.org/pkg/net/url/#Values.Get

Reply:
Done